core/src/main/scala/org/apache/spark/eventhubs/client/EventHubsClient.scala

```diff
@@ -222,45 +225,38 @@ private[spark] class EventHubsClient(private val ehConf: EventHubsConf)
         synchronized(needsTranslation += nAndP)
       }
     }
-
     logInfo(s"translate: needsTranslation = $needsTranslation")
-    val consumerGroup = ehConf.consumerGroup.getOrElse(DefaultConsumerGroup)
 
+    val consumerGroup = ehConf.consumerGroup.getOrElse(DefaultConsumerGroup)
     val threads = ArrayBuffer[Thread]()
     needsTranslation.foreach(nAndP => {
       val partitionId = nAndP.partitionId
       threads += new Thread {
         override def run(): Unit = {
-          val receiver = client
-            .createReceiverSync(consumerGroup,
-                                partitionId.toString,
-                                positions.getOrElse(nAndP, defaultPos).convert)
-          receiver.setPrefetchCount(PrefetchCountMinimum)
-          val events = receiver.receiveSync(1) // get the first event that was received.
-          receiver.closeSync()
-          if (events == null || !events.iterator().hasNext) {
-            // no events to receive at EndOfStream can happen in 2 cases
-            // 1. receive at endOfStream and no new events arrive
-            // 2. receive on a brand new partition
-            val receiverOptions = new ReceiverOptions()
-            receiverOptions.setReceiverRuntimeMetricEnabled(true)
-            val newReceiver = client
+          @volatile var receiver: PartitionReceiver = null
+          try {
+            receiver = client
               .createReceiverSync(consumerGroup,
                                   partitionId.toString,
-                                  com.microsoft.azure.eventhubs.EventPosition.fromStartOfStream,
-                                  receiverOptions)
-            val startEvents = newReceiver.receiveSync(1)
-            if (startEvents == null || !startEvents.iterator().hasNext) {
-              result.put(partitionId, StartingSequenceNumber)
-            } else {
-              val lastSequenceNumber =
-                newReceiver.getRuntimeInformation.getLastEnqueuedSequenceNumber
-              result.put(partitionId, lastSequenceNumber + 1)
-            }
-            newReceiver.closeSync()
-          } else {
+                                  positions.getOrElse(nAndP, defaultPos).convert)
+            receiver.setPrefetchCount(PrefetchCountMinimum)
+            receiver.setReceiveTimeout(Duration.ofSeconds(5))
+            val events = receiver.receiveSync(1) // get the first event that was received.
             val event = events.iterator.next
```

Comment:
>val event = events.iterator.next [](start = 12, length = 32)

build safe gaurd for; if receiveSync returns null events.iterator would throw nullpointerex.

Reply:
The intention here is to generate a NPE which should only happen from a receiver timeout. 

I was considering breaking up the catch block like so:

```scala
catch {
  case a: NullPointerException => 
    // handle null event case
  case b: IllegalEntityException =>
    // stop job
```
Thoughts on that approach?

Reply:
I would recommended using `if else` as the control-flow logic than `try catch` ! :)
as we can simply check events == null - letting it throw `nullpointerex` is a bit counter intuitive :)

Reply:
Ok, cool. So, it'll end up like this:

```scala
try {
  // try to receive an event, if it's null handle that accordingly
} catch {
  // explicitly catch IllegalEntityException, educate customer on consumer groups, kill the job
} finally { 
  // clean up 
}
```

Look good?